### PR TITLE
Support lists (or collections) for Node's name property

### PR DESCRIPTION
**Pull request by chrissng**
A person may have more than one name. For instance an English and a Chinese name. As such a person, represented as a Node type may have name property represented as a list. Implemented functionality to ensure lists, tuples and sets are supported for Node's name property. Some unit testing to cover this functionality.

p.s. There's a PR previously just before author announced the project won't be maintained

### CODE REVIEW

File: py2neo/util.py [hunk 1]
```diff
@@ -33,6 +33,9 @@
 
 
 def snake_case(s):
+    if is_collection(s):
+        s = "_".join(s)
```
Comment (technige) [inline py2neo/util.py]: This would export unexpected behaviour from the snake_case function. It would be better to perform joining _outside_ the function.

**Comment by technige** [comment]
When can you see this being used? The `__name__` attribute is intended purely for use by `repr` for human readability. It serves no real purpose from a data storage point of view.

**Comment by chrissng** [comment]
Understand that `property['name']` is used by `repr` for human readability. It should also be okay that names be stored as list in neo4j. So the list will be converted into a string before running the snake_case.

**Comment by technige** [comment]
Closing as wontfix.

**Pull request closed without merging**